files/image_config/platform/rc.local

```diff
@@ -29,6 +104,44 @@ if [ ! -e /host/machine.conf ]; then
         done
     fi
 
+    # Extract the previous NOS's partition that contains the migration artifacts
+    set -- $(cat /proc/cmdline)
+    for x in "$@"; do
+        case "$x" in
+            nos-config-part=*)
+                nos_val="${x#nos-config-part=}"
+                ;;
+        esac
+    done
+
+    if [ -n "$nos_val" ]; then
+        nos_dev=$(findfs $nos_val)
+        if [ $? != 0 ]; then
+            log_migration "ERROR: nos_dev not found. Check grub parameters"
+        fi
+    else
+        log_migration "ERROR: nos_val not found. Check grub parameters"
+    fi
+
+    if [ -n "$nos_dev" ]; then
+        # Mount the previous NOS's partition
+        mkdir -p /mnt/nos_migration
+        mount $nos_dev /mnt/nos_migration
+        mkdir -p /host/fast-reboot
+
+        # Copy relevant files
+        nos_migration_import /mnt/nos_migration/mgmt_interface.cfg /host/migration
+        nos_migration_import /mnt/nos_migration/minigraph.xml /host/migration
+        nos_migration_import /mnt/nos_migration/arp.json /host/fast-reboot
+        nos_migration_import /mnt/nos_migration/fdb.json /host/fast-reboot
+        nos_migration_import /mnt/nos_migration/default_routes.json /host/fast-reboot
+
+        umount /mnt/nos_migration
+        rmdir /mnt/nos_migration
+    fi
+
+    update_mgmt_interface_macaddr /host/migration/mgmt_interface.cfg
```

Comment:
Are we sure this will work for any platform?

Reply:
There is an assumption that the platform supports ethtool to modify the eth (GbE in S6100/Z9100) case using igb. If the platform does not support ethtool to change the EEPROM, the alternative (workaround) is to note that the unit has migrated and apply the previous NOS' mac to eth0 (using ip link set) everytime the unit reboots...